Project-Aurora/Project-Aurora/Devices/Razer/RazerDevice.cs

```diff
@@ -92,16 +116,16 @@ protected override bool UpdateDevice(Dictionary<DeviceKeys, System.Drawing.Color
                 keypad.Set(ToColore(clr));
             }
 
-            foreach (var key in keyColors)
+            foreach (var key in keyColors.ToList())
```

Comment:
is list duplication really necessary? This will be called many times each second. I guess getting error while changing layouts is an okay drawback to gain this optimization